pkg/platform/coverage/java/utils.go

```diff
@@ -17,12 +17,19 @@ import (
 	"go.uber.org/zap"
 )
 
-func downloadAndExtractJaCoCoCli(logger *zap.Logger, version, dir string) error {
-	cliPath := filepath.Join(dir, "jacococli.jar")
+func DownloadAndExtractJaCoCoCli(logger *zap.Logger) error {
+	jacocoPath := filepath.Join(os.TempDir(), "jacoco")
+	err := os.MkdirAll(jacocoPath, 0777)
+	if err != nil {
+		logger.Debug("failed to create jacoco directory", zap.Error(err))
+		return err
+	}
+
+	cliPath := filepath.Join(jacocoPath, "jacococli.jar")
 
-	downloadURL := fmt.Sprintf("https://github.com/jacoco/jacoco/releases/download/v%s/jacoco-%s.zip", version, version)
+	downloadURL := fmt.Sprintf("https://github.com/jacoco/jacoco/releases/download/v%s/jacoco-%s.zip", "0.8.12", "0.8.12")
 
```

Comment:
Any specific reason to hard code the version ?? Can we fetch the latest ?

Reply:
this was the latest at that time, so I used it. fetching latest release through mvn api's is possible, but i may increase time, so i hardcoded a  specific version. also hardcoding may ensure If jacoco releases a breaking change, the coverage won't break.